doc/docs.md

```diff
@@ -1331,6 +1331,45 @@ println(numbers) // [1, 2, 3]
 ```
 When an identifier is just a single underscore, it is ignored.
 
+##### Custom iterators
+
+```v
+struct SquareIterator {
+	arr []int
+mut:
+	idx int
+}
+
+fn (mut iter SquareIterator) next() ?int {
+	if iter.idx >= iter.arr.len {
+		return error('')
+	}
+
+	item := iter.arr[iter.idx]
+	iter.idx++
+	return item * item
+}
+
+nums := [1, 2, 3, 4, 5]
+iter := SquareIterator{
+	arr: nums
+}
+for squared in iter {
+	println(squared)
+}
+
+```
+
+Types that implement a `next` method returning an `Option` can be iterated
```

Comment:
I would move this paragraph up under the header (so they know what the code is for before they try to read through it),

Reply:
alright i can move it below